src/devices/xyzroe.ts

```diff
@@ -463,6 +483,117 @@ const definitions: DefinitionWithExtend[] = [
+            await endpoint.configureReporting('haElectricalMeasurement', [
+                {
+                    attribute: 'activePower',
+                    minimumReportInterval: CONSTANTS.MIN_REPORT_INTERVAL,
+                    maximumReportInterval: CONSTANTS.MAX_REPORT_INTERVAL,
+                    reportableChange: CONSTANTS.REPORTABLE_CHANGE,
+                },
+            ]);
+
+            await endpoint.read('haElectricalMeasurement', ['acCurrentMultiplier', 'acCurrentDivisor']);
+            await endpoint.configureReporting('haElectricalMeasurement', [
+                {
+                    attribute: 'rmsCurrent',
+                    minimumReportInterval: CONSTANTS.MIN_REPORT_INTERVAL,
+                    maximumReportInterval: CONSTANTS.MAX_REPORT_INTERVAL,
+                    reportableChange: CONSTANTS.REPORTABLE_CHANGE,
+                },
+            ]);
+
+            await endpoint.read('haElectricalMeasurement', ['acVoltageMultiplier', 'acVoltageDivisor']);
+            await endpoint.configureReporting('haElectricalMeasurement', [
+                {
+                    attribute: 'rmsVoltage',
+                    minimumReportInterval: CONSTANTS.MIN_REPORT_INTERVAL,
+                    maximumReportInterval: CONSTANTS.MAX_REPORT_INTERVAL,
+                    reportableChange: CONSTANTS.REPORTABLE_CHANGE,
+                },
+            ]);
+        },
+        exposes: [
+            e.binary('alarm', ea.STATE, true, false).withDescription('🔌 Indicates if overcurrent protection is triggered').withEndpoint('1'),
+            e
+                .power_on_behavior(['off', 'on', 'toggle', 'previous'])
+                .withDescription('🔌 Controls the behavior when the device powers on after a power loss')
+                .withEndpoint('1'),
+            e
+                .binary('state', ea.ALL, 'ON', 'OFF')
+                .withProperty('state')
+                .withValueToggle('TOGGLE')
+                .withDescription('🔌 Controls the USB port switch')
+                .withEndpoint('1'),
+            e
+                .binary('state', ea.ALL, 'ON', 'OFF')
+                .withProperty('state')
+                .withValueToggle('TOGGLE')
+                .withDescription('💡 Indicates the Zigbee status via LED')
+                .withEndpoint('2'),
+            e
+                .binary('state', ea.ALL, 'ON', 'OFF')
+                .withProperty('state')
+                .withValueToggle('TOGGLE')
+                .withDescription('💡 Indicates the USB state via LED')
+                .withEndpoint('3'),
+        ],
+        extend: [
+            deviceEndpoints({endpoints: {1: 1, 2: 2, 3: 3}}),
+            identify(),
+            electricityMeter({
+                cluster: 'electrical',
+                configureReporting: false,
+                endpointNames: ['1'],
+            }),
+            temperature({
+                reporting: {
```

Comment:
Why use custom reporting values here?

Reply:
For a more positive user experience, I would like to have a discomfort setting for reponding like this.